src/learning/lda.rs

```diff
@@ -0,0 +1,250 @@
+        let mut rng = thread_rng();
+        for (document, row) in input.row_iter().enumerate() {
+            let mut document_topics = Vec::with_capacity(row.sum());
+            for (word, word_count) in row.iter().enumerate() {
+                for _ in 0..*word_count{
+                    let topic = rng.gen_range(0, topic_count);
+                    result.document_topic_count[[document, topic]] += 1.0;
+                    result.topic_total_by_document[document] += 1.0;
+                    result.topic_word_count[[topic, word]] += 1.0;
+                    result.word_total_by_topic[topic] += 1.0;
+                    document_topics.push(topic);
+                }
+            }
+            topics.push(document_topics);
+        }
+        (result, topics)
+    }
+
+    /// Find the distribution of words over topics.  This gives a matrix where the rows are
+    /// topics and the columns are words.  Each entry (topic, word) gives the probability of
+    /// word given topic.
+    pub fn phi(&self) -> Matrix<f64> {
+        let mut distribution = self.topic_word_count.clone() + self.beta;
+        let row_sum = distribution.sum_rows();
+        // XXX To my knowledge, there is not presently a way in rulinalg to divide a matrix
+        // by a vector, so I do it with a loop here.
+        for (mut row, sum) in distribution.row_iter_mut().zip(row_sum.iter()) {
+            *row /= sum;
+        }
+        distribution
+    }
+}
+
+impl LDA {
+    fn conditional_distribution(&self, result: &LDAResult, document: usize, word: usize, vocab_count: f64, topic_count: f64) -> Vector<f64> {
+
+        // Convert the column of the word count by topic into a vector
+        let word_topic_count:Vector<f64> = result.topic_word_count.col(word).into();
+
+        // Calculate the proportion of this word's contribution to each topic
+        let left:Vector<f64> = (word_topic_count + self.beta).elediv(
+            &(result.word_total_by_topic.clone() + self.beta * vocab_count)
+        );
+
+        // Convert the row of the topic count by document into a vector
+        let topic_document_count:Vector<f64> = result.document_topic_count.row(document).into();
+
+        // Calculate the proportion of each topic's contribution to this document
+        let right:Vector<f64> =  (topic_document_count + self.alpha) /
+            (result.topic_total_by_document[document] + self.alpha * topic_count);
+
+        // Multiply the proportions together to this word's contribution to the document by topic
+        let mut probability:Vector<f64> = left.elemul(&right);
+
+        // Normalize it so that it's a probability
+        probability /= probability.sum();
+        return probability;
+    }
+}
+
+impl UnSupModel<(Matrix<usize>, usize), LDAResult> for LDA {
+    /// Predict categories from the input matrix.
+        fn predict(&self, inputs: &(Matrix<usize>, usize)) -> LearningResult<LDAResult> {
```

Comment:
I think that we need to be clearer about what these inputs are. I am assuming that the first tuple argument is the word counts per document, and the second is the number of iterations?

It would follow the rest of the library more closely if the iterations were specified at the model level, and this function simply took the `&Matrix<usize>`.

And one other small note, in order for docs to show up with rustdoc they must be placed on the outside of the impl block. e.g.

```rust
/// Doc here will be rendered
impl UnSupModel<..> ... {
    /// Docs here will not
    fn predict(...) -> ... { }
}
```